onnxruntime/core/providers/cpu/llm/attention_helper.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "core/common/common.h"
+#include "core/providers/common.h"
+
+namespace onnxruntime {
+namespace attention_helper {
+
+enum AttentionMaskType {
+  MASK_NONE,  // No mask
+  MASK_BOOL,  // Boolean mask with shape [batch_size, sequence_length, total_sequence_length] or [batch_size, q_num_heads, q_sequence_length, total_sequence_length
+  MASK_ADD,   // Additive mask with shape [batch_size, sequence_length, total_sequence_length] or [batch_size, q_num_heads, q_sequence_length, total_sequence_length
+};
+
+enum AttentionType {
+  kInvalid = 0,              // Invalid attention type
+  kMultiHeadAttention = 1,   // Multi-headed attention (MHA)
+  kGroupQueryAttention = 2,  // Group query attention (GQA)
+  kMultiQueryAttention = 3,  // Multi-query attention (MQA)
+};
+
+enum QKMatMulOutputMode {
```

Comment:
Shall we have a value for not output?

Reply:
In that case, the onnx models only specifies 3 outputs at most. The pointer for this output is null.

Reply:
While it is not necessary for this opset 23 version, I think it would be useful to have a value for "no output", may be -1. We should fix this in ONNX 24 spec.

Reply:
I understand that we can check the output to see whether it need output. I mean, add a enum value could make the behavior consistent to avoid confusion.